part-customization.tex

```diff
@@ -43,7 +43,156 @@ \section{Introduction}
 
 Another example is changing ``Workflow configuration''\index{workflow configuration}, by which the life cycle of invoices and GL transactions can be changed.  This way, additional steps can be introduced into the life cycle, or steps can be combined (from a user perspective) by automatically executing steps after another has been triggered manually.
 
+Customization usually means additional or modified Perl code needs to be loaded. 
+
+@@@TODO: describe where to store this new or overriding code...
+
+\section{Workflow configuration}
+
+LedgerSMB uses the concept of workflows\index{workflows} to manage the document life cycle.
+An invoice for example may be \texttt{saved}, \texttt{posted} or \texttt{reversed}.  These
+are states in the life cycle of the invoice.  Each state may have associated actions.  In
+case of the invoice an example may be to \texttt{Post} it when it's \texttt{saved} or to
+\texttt{E-mail} it when it's \texttt{posted}.
+
+Actions associated with a state may be available or not determined by one or more conditions.
+An examples are ``is the posting date of the transaction in a closed period''
+and ``is the current user allowed to post transactions''.
+
+Workflows are stored in the \texttt{workflows/} directory.  Each workflow consists of
+several files, prefixed with the name of the workflow (e.g. \texttt{ar-ap.}):
+
+\begin{enumerate}
+	\item [\texttt{actions.xml}] (optional) names the list of available transitions
+	in the workflow naming Perl code to be executed
+	\item [\texttt{conditions.xml}] (optional) names the list of conditions to be used
+	in the workflow naming Perl code to evaluate
+	\item [\texttt{persisters.xml}] (optional) defines the Perl mechanism to load and store
+	workflow instances
+	\item [\texttt{workflow.xml}] (required) defines the life cycle (states, actions
+	and conditions) with the allowed transitions per state (actions).
+\end{enumerate}
+
+Customized versions of workflow definition files must be stored in
+ \texttt{./customized\_workflows/}\footnote{See
+ 	\secref{subsubsec-global-config-ledgersmb-yaml-workflows} on how to configure
+ these paths}.  Files stored in this location will be used to override the ones in
+the standard location \texttt{./workflows}.  No changes should be made directly to
+the standard files because of the risk that those files will be overwritten on upgrade
+-- loosing the changes.

```

Comment:
How is are the standard workflow files disabled?  The use case is providing a custom workflow to replace a system provided workflow.

Reply:
Do you think the expanded explanation makes sense?